pkg/oauthflow/device.go

```diff
@@ -173,3 +195,46 @@ func (d *DeviceFlowTokenGetter) GetIDToken(p *oidc.Provider, cfg oauth2.Config)
 		Subject:   subj,
 	}, nil
 }
+
+// CodeURL fetches the device authorization endpoint URL from the provider's well-known configuration endpoint
+func (d *DeviceFlowTokenGetter) CodeURL() (string, error) {
+	if d.codeURL != "" {
+		return d.codeURL, nil
+	}
+
+	wellKnown := strings.TrimSuffix(d.Issuer, "/") + "/.well-known/openid-configuration"
```

Comment:
Is this token authenticated at this point? Otherwise, it could be used as a DOS vector for making a request to an untrusted website.

Reply:
By this point, a lower level method within the oidc library has already fetched this information and verified that it matches the intended provider. It just does not persist all the information in a usable way, so we need to re-fetch it here.